README.md

```diff
@@ -0,0 +1,268 @@
+mmap(va:0x8000181000, 323584, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE|MAP_FIXED, fd:3, 1556480) = va:0x8000181000
+mmap(va:0x80001d0000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_DENYWRITE|MAP_FIXED, fd:3, 1875968) = va:0x80001d0000
+mmap(va:0x80001d6000, 13952, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_FIXED, fd:-1, 0) = va:0x80001d6000
+close(fd:3) = OK
+```
+In this case, the virtual base address of `libc.so.6` is `0x8000005000`.
+
+To load the application and library symbols appropriately, as described in the
+previous subsection, you must add the segment offset of the `.text` section to
+virtual base address. This allows you to load the symbols from `gdb` with
+`add-symbol-file`.
+
+#### Recommendation: `gdb` Setup Script
+Because of the calculations, we recommend scripting the `gdb` setup so that any
+subsequent `gdb` debug session will be ready quickly. You can get inspiration
+from the following `bash` script, which provides a function that automatically
+determines the `.text` offset and adds it to a given base address.
+The advantage is that only the base addresses have to be noted from the
+Unikraft console output.
+
+```bash
+#!/bin/bash
+# Host and port of the GDB server port (qemu)
+GDBSRV=":1234"
+
+# Generate a GDB command line for loading the symbols of an ELF executable/
+# shared library.
+# Usage: gdb-add-symbols "<ELF executable/library>" \
+#                        "<base load address (hex, no leading '0x')>"
+gdb-add-symbols()
+{
+	local LOAD_ELF="$1"
+	local LOAD_ADDR="${2}"
+	local LOAD_TADDR=
+	local TEXT_OFFSET=
+
+	# Hacky way to figure out the .text offset
+	TEXT_OFFSET=$( readelf -S "${LOAD_ELF}" | grep '.text' | awk '{ print $5 }' )
+
+	# Compute offset of .text section with base address
+	LOAD_TADDR=$( printf 'obase=16;ibase=16;%s+%s\n' "${LOAD_ADDR^^}" "${TEXT_OFFSET^^}" | bc )
+
+	# Generate GDB command
+	printf 'add-symbol-file -readnow %s 0x%s' "${LOAD_ELF}" "${LOAD_TADDR}"
+}
+
+# Connect to $GDBSRV and set up gdb
+# NOTE: The first block of instructions connects to the gdb port of the qemu
+#       process and follows the CPU mode change while the guest is booting.
+#       This is currently a requirement for using gdb with qemu for x86_64. For
+#       other platforms and architectures, this may be different.
+#       This block assumes that the guest is started in paused state (`-P` for
+#       `qemu-guest`).
+# HINT: You can use the `directory` command to specify additional paths that
+#       `gdb` will use to search for source files.
+#       For example, if you run your dynamically linked application with
+#       Debian's libc, you can install (`apt install glibc-source`) and
+#       extract the glibc sources under /usr/src/glibc.
+#         --eval-command="directory /usr/src/glibc/glibc-2.31"
+exec gdb \
+	--eval-command="target remote $GDBSRV" \
+	--eval-command="hbreak _libkvmplat_entry" \
```

Comment:
This symbol name was removed in https://github.com/unikraft/unikraft/pull/722.

Probably it should be changed to `_multiboot_entry` or `_ukplat_entry`.

Reply:
What @csvancea said.